### ISSUE TITLE
Use semantic versioning please!

### ISSUE THREAD

**Issue by nemoinho** [valor-software/ngx-bootstrap#1868]
tl;dr: The broken versioning breaks npm compatibility, so switch to semantic versioning and follow the [npm standards](https://docs.npmjs.com/getting-started/semantic-versioning)

I saw several flaws in the development process, that the project don't follow semantic versioning. This issue prevents users from the use of the caret notation (e.g ^1.2.3), which is default for npm install --save.

As an example for this wrong understanding I will show the changes in dropdown between version 1.4 and 1.5 ([changes](https://github.com/valor-software/ng2-bootstrap/commit/31cba418a13674917e0fcfba2419c7f27c660ba1))
These changes destroy the compatibility with the previous Version and require a major change therefore.

Reminder:
Semantic Versioning consists of three parts, MAJOR, MINOR and PATCH these parts mean:
- MAJOR: incompatible change backward compatibility breaks like an API-change (see example above)
- MINOR: compatible change like adding another feature
- PATCH: any kind of fix for the implementation against the defined API

**Comment by markpeterfejes** [comment]
This would be really cool, indeed. Same as #1838

**Comment by valorkin** [comment]
I will do, sorry again

**Comment by nemoinho** [comment]
By the way versions are for computers, names are for human, so don't regret to increase the version if you change something. Don't increasing numbers where you should do is a pain for all developers which trust you.
